presto-main/src/main/java/com/facebook/presto/sql/planner/PlanOptimizers.java

```diff
@@ -89,6 +89,7 @@
 import com.facebook.presto.sql.planner.iterative.rule.TransformCorrelatedLateralJoinToJoin;
 import com.facebook.presto.sql.planner.iterative.rule.TransformCorrelatedScalarAggregationToJoin;
```

Comment:
Divide this PR to 2 commits, one with just migration and the other with adding support for query mentioned in the test.

Reply:
It is slightly hard since the changes are deeply ingrained with each other. Would you mind reviewing this one as is? I will take care of this in further PRs

Reply:
It is not about me, it is more about git history. Having this is in separate commits has many benefits. Imagine that you introduce a bug, like InlineProjections does not work well with subquery rule for some cases we do not know now. In such case we could revert only that change and leave migration as it is.

Reply:
Ok, got that. I will rewrite this commit to split accordingly.